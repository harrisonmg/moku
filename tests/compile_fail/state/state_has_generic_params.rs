use moku::*;

#[state_machine]
mod blinky {
    use moku::*;

    #[machine_module]
    mod state_machine {}

    struct Top;
    impl TopState<BlinkyState> for Top {}

    struct Bottom<T> {
        t: T,
    }

    #[superstate(Top)]
    impl<T> State<BlinkyState> for Bottom<T> {}
}

fn main() {}
