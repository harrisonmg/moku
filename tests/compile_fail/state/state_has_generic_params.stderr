error: `moku::State`s must not have generic parameters
  --> tests/compile_fail/state/state_has_generic_params.rs:18:36
   |
18 |     impl<T> State<BlinkyState> for Bottom<T> {}
   |                                    ^^^^^^
